### PR TITLE
feat: nav updates

### LINKED ISSUE #9970: Implement dash nav design updates
## New

image

## Current

image

This issue is to implement updates to the nav design per recent internal discussion.

### Enhancement
- Removes the borders, tier tag, nav item for settings and members, and menu from the current version
- Adds the updated organization name section label, and the icon button for settings and members with tooltip
- Refines margins etc.
- Stretch: mobile changes TBD in this or a separate issue

### Estimated effort
- 3 days (total implementation, testing, deploying)

### PR DESCRIPTION
**Pull request by ackernaut**
Closes #9970 

[Demo](https://www.loom.com/share/05f7b9971a3e48cf86de362af7d89b87) and [on-demand env](https://nav-updates.parabol.fun/meetings)

### Test
- [x] Nav follows UI designs for dash nav and org admin nav for both desktop and mobile
- [x] Org nav sections are sorted by tier (enterprise, team, starter) then alphanumeric within each tier

## Summary by CodeRabbit

- **New Features**
  - Added tooltip functionality for organization actions in the dashboard navigation list.
  
- **Style**
  - Updated styling across multiple components to improve visual consistency and user experience.
  - Refined padding, font styles, and layout properties for better presentation.

- **Refactor**
  - Streamlined imports and replaced several styled components to enhance code maintainability.
  - Improved rendering logic for better organisation display and interaction.

### CODE REVIEW

File: packages/client/components/DashNavList/DashNavList.tsx [hunk 1]
```diff
@@ -56,44 +42,42 @@ const DashNavList = (props: Props) => {
     `,
     organizationsRef
   )
-  const isDesktop = useBreakpoint(Breakpoint.SIDEBAR_LEFT)
-  const teams = organizations?.flatMap((org) => org.viewerTeams)
+
+  const tiers = ['enterprise', 'team', 'starter']
```
Comment (nickoferrall) [inline packages/client/components/DashNavList/DashNavList.tsx:46]: -1 we should use `TierEnum` here instead as the we might change the names, e.g. `team` back to `pro`

**Review by nickoferrall — CHANGES_REQUESTED** [review]
Looks good to me! I think the `TierEnum` array needs changing, but other than that, I think it's ready to be merged. 

I prefer this design. It looks a lot cleaner, and the settings icon really gets my attention with the lighter team names.

File: packages/client/components/DashNavList/DashNavList.tsx (hunk 1 above)
Comment (ackernaut) [reply packages/client/components/DashNavList/DashNavList.tsx:46]: I see this in other places, but I’m not sure the best way to import and define a `TierEnum` here.

File: packages/client/components/DashNavList/DashNavList.tsx [hunk 2]
```diff
@@ -43,11 +44,11 @@ const DashNavList = (props: Props) => {
     organizationsRef
   )
 
-  const tiers = ['enterprise', 'team', 'starter']
+  const TierEnumValues: TierEnum[] = ['enterprise', 'team', 'starter']
```
Comment (ackernaut) [inline packages/client/components/DashNavList/DashNavList.tsx:47]: @nickoferrall does this work to check the tiers?

Comment (ackernaut) [reply packages/client/components/DashNavList/DashNavList.tsx:47]: See [commit](https://github.com/ParabolInc/parabol/pull/9973/commits/ff294be2931c17ecb3832b55e1762113ebe88d14)

Comment (nickoferrall) [reply packages/client/components/DashNavList/DashNavList.tsx:47]: That works!

**Review by nickoferrall — APPROVED** [review]

**Pull request merged**